pkg/controller/localvolumeset/reconcile.go

```diff
@@ -0,0 +1,133 @@
+	r.reqLogger = logf.Log.WithName(controllerName).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.reqLogger.Info("Reconciling LocalVolumeSet")
+	// Fetch the LocalVolumeSet instance
+	instance := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.lvSetMap.deregisterLocalVolumeSet(instance.Spec.StorageClassName, request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		r.reqLogger.Error(err, "failed to get localvolumeset")
+		return reconcile.Result{}, err
+	}
+
+	// store a one to many association from storageClass to LocalVolumeSet
+	r.lvSetMap.registerLocalVolumeSet(instance.Spec.StorageClassName, request.NamespacedName)
+
+	// The diskmaker daemonset, local-staic-provisioner daemonset and configmap are created in pkg/daemon
+	// this way, there can be one daemonset for all LocalVolumeSets
+
+	err = r.syncStorageClass(instance)
+	if err != nil {
+		r.reqLogger.Error(err, "failed to sync storageclass")
+		return reconcile.Result{}, err
+	}
+
+	r.reqLogger.Info("updating status")
+	err = r.updateStatus(instance)
+	if err != nil {
+		r.reqLogger.Error(err, "failed to update status")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *LocalVolumeSetReconciler) updateStatus(lvs *localv1alpha1.LocalVolumeSet) error {
+	// fetch PVs that match the storageclass
+	pvs := &corev1.PersistentVolumeList{}
+	err := r.client.List(context.TODO(), pvs, client.MatchingFields{pvStorageClassField: lvs.Spec.StorageClassName})
+	if err != nil {
+		return err
+	}
+	totalPVCount := int32(len(pvs.Items))
+	lvs.Status.TotalProvisionedDeviceCount = &totalPVCount
+	lvs.Status.ObservedGeneration = lvs.Generation
+	err = r.client.Status().Update(context.TODO(), lvs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *LocalVolumeSetReconciler) syncStorageClass(lvs *localv1alpha1.LocalVolumeSet) error {
+	deleteReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	firstConsumerBinding := storagev1.VolumeBindingWaitForFirstConsumer
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      lvs.Spec.StorageClassName,
+			Namespace: lvs.GetNamespace(),
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, storageClass, func() error {
+		if storageClass.CreationTimestamp.IsZero() {
+			storageClass.Provisioner = "kubernetes.io/no-provisioner"
+			storageClass.ReclaimPolicy = &deleteReclaimPolicy
+			storageClass.VolumeBindingMode = &firstConsumerBinding
+			storageClass.ObjectMeta.Labels = map[string]string{
+				util.OwnerNameLabel:      lvs.GetName(),
```

Comment:
non-namespaced objects can't hold referenced to namespaced objects - https://github.com/openshift/local-storage-operator/pull/114/files it results in hard to track bugs.

Reply:
Had already removed in unpushed version :grimacing:

Reply:
fixed